modules/test/playwright/tests/asset-tags-admin-web/tagsUseCases.spec.ts

```diff
@@ -0,0 +1,224 @@
+
+	await apiHelpers.headlessDelivery.postStructuredContent({
+		contentStructureId,
+		datePublished: null,
+		siteId: site.id,
+		tags: ['first', 'second'],
+		title: getRandomString(),
+		viewableBy: 'Anyone',
+	});
+
+	await apiHelpers.headlessDelivery.postStructuredContent({
+		contentStructureId,
+		datePublished: null,
+		siteId: site.id,
+		tags: ['first'],
+		title: getRandomString(),
+		viewableBy: 'Anyone',
+	});
+
+	await pageEditorPage.goto(layout, site.friendlyUrlPath);
+	await pageEditorPage.goToWidgetConfiguration(widgetId);
+
+	await tagsFilterPage.changeDisplayTemplate('Number');
+	await tagsFilterPage.configurationIframe
+		.getByLabel('Show Asset Count')
+		.check();
+	await tagsFilterPage.saveAndClose();
+
+	await pageEditorPage.publishPage();
+
+	await page.goto(`/web/${site.name}/${layout.friendlyUrlPath}`);
+
+	await expect(
+		await page.getByRole('link', {name: 'first (2)'})
+	).toBeVisible();
+	await expect(
+		await page.getByRole('link', {name: 'second (1)'})
+	).toBeVisible();
+
+	test.step('Tags can be increased', async () => {
+		await apiHelpers.headlessDelivery.postStructuredContent({
+			contentStructureId,
+			datePublished: null,
+			siteId: site.id,
+			tags: ['first'],
+			title: getRandomString(),
+			viewableBy: 'Anyone',
+		});
+
+		await page.goto(`/web/${site.name}/${layout.friendlyUrlPath}`);
+
+		await expect(
+			await page.getByRole('link', {name: 'first (3)'})
+		).toBeVisible();
+		await expect(
+			await page.getByRole('link', {name: 'second (1)'})
+		).toBeVisible();
+	});
+});
+
+test('Add an auto assertion tag via blogs widget', async ({
+	apiHelpers,
+	page,
+	pageEditorPage,
+	site,
+}) => {
+	const widgetId = getRandomString();
+	const widgetDefinition = await getWidgetDefinition({
+		id: widgetId,
+		widgetName: 'com_liferay_blogs_web_portlet_BlogsPortlet',
+	});
+	const layout = await apiHelpers.headlessDelivery.createSitePage({
+		pageDefinition: getPageDefinition([widgetDefinition]),
+		siteId: site.id,
+		title: widgetId,
+	});
+
+	const blogsEntryName = getRandomString();
+
+	await apiHelpers.headlessDelivery.postBlog(site.id, {
+		headline: blogsEntryName,
+		keywords: ['first', 'second', 'third'],
+	});
+
+	await pageEditorPage.goto(layout, site.friendlyUrlPath);
+
+	await pageEditorPage.publishPage();
+
+	await page.goto(`/web/${site.name}/${layout.friendlyUrlPath}`);
+
+	await page.getByRole('link', {name: blogsEntryName}).click();
+
+	await expect(await page.getByText('first second third')).toBeVisible();
+
+	await test.step('Tag can be removed', async () => {
+		await page
+			.getByRole('link', {name: 'Edit Entry'})
+			.evaluate((element: HTMLElement) => element.click());
+
+		await page.getByRole('button', {name: 'Categorization'}).waitFor();
+
+		await expect(async () => {
+			await page.getByRole('button', {name: 'Categorization'}).click();
+
+			await expect(page.getByLabel('Remove first')).toBeVisible();
+		}).toPass();
+
+		await page.getByLabel('Remove first').click();
```

Comment:
Isn't it better to delete the middle ('second') option and then check that the deletion is correct `getByText('first third')`?

Reply:
Why it's worst to remove the first? The expect() block is there to make sure we load the modal in before we try to remove the tag.. Without it the test was flaky